Domains/1-Common/DistributionSystems/DistributionSystems.ecschema.xml

```diff
@@ -62,4 +62,83 @@
+            <ECCustomAttributes>
+                <HiddenProperty xmlns="CoreCustomAttributes.01.00.03"/>
+            </ECCustomAttributes>
+        </ECProperty>
+    </ECEntityClass>
+    
+    <ECRelationshipClass typeName="DistributionElementOwnsDistributionPorts" strength="embedding" modifier="None" description="A relationship that relates child distribution ports to parent distribution element.">
+        <BaseClass>bis:ElementOwnsChildElements</BaseClass>
+        <Source multiplicity="(0..1)" roleLabel="owns" polymorphic="true">
+            <Class class="IDistributionElement"/>
+        </Source>
+        <Target multiplicity="(0..*)" roleLabel="is owned by" polymorphic="true">
+            <Class class="DistributionPort"/>
+        </Target>
+    </ECRelationshipClass>
+
+    <ECEntityClass typeName="PortConnection" modifier="None" displayLabel="Port Connection" description="An element that indicates a connection between 2 distribution ports.">
+        <BaseClass>bis:InformationRecordElement</BaseClass>
+    </ECEntityClass>
+
+    <ECRelationshipClass typeName="PortConnectionConnectsPorts" strength="referencing" modifier="Abstract" description="Indicates that the DistributionPort is connected to some other Distribution ports via PortConnection.">
+        <BaseClass>bis:ElementRefersToElements</BaseClass>
+        <Source multiplicity="(0..1)" roleLabel="connects" polymorphic="true">
+            <Class class="PortConnection"/>
+        </Source>
+        <Target multiplicity="(0..*)" roleLabel="is connected by" polymorphic="true">
+            <Class class="DistributionPort"/>
+        </Target>
+    </ECRelationshipClass>
+
+    <ECRelationshipClass typeName="PortConnectionConnectsTwoPorts" strength="referencing" modifier="Abstract" description="Indicates that the DistributionPort is connected to one other Distribution port via PortConnection.">
+        <BaseClass>PortConnectionConnectsPorts</BaseClass>
+        <Source multiplicity="(0..1)" roleLabel="connects" polymorphic="true">
+            <Class class="PortConnection"/>
+        </Source>
+        <Target multiplicity="(0..2)" roleLabel="is connected by" polymorphic="true">
+            <Class class="DistributionPort"/>
+        </Target>
+    </ECRelationshipClass>
+    
+    <ECRelationshipClass typeName="PortConnectionIsRealizedByPhysicalElements" strength="referencing" modifier="None" description="Indicates that the Port Connection is realized by some Physical Element(s).">
+        <BaseClass>bis:ElementRefersToElements</BaseClass>
+        <Source multiplicity="(0..1)" roleLabel="is realized by" polymorphic="true">
+            <Class class="PortConnection"/>
+        </Source>
+        <Target multiplicity="(0..*)" roleLabel="realizes" polymorphic="true">
+            <Class class="bis:PhysicalElement"/>

```

Comment:
Would it make sense to change Target constraint to GeoemtricElement3d? To enable use by non-PhysicalElements, e.g. Analytical, or perhaps even Functional?

Reply:
we should be able to inject a baseclass relationship later. 
so we would still have this particular relationship with constraints as defined here, 

Furthermore, rather than this single relationship, in which case we would also need a base class for DistributionPort and PortConnection, in some now schema.. Maybe we could call it ElementConnectivity. which brings us to a whole new discussion, so best defer it for later.

do you have a sample use case in mind where we would have analytical elements connected by ports?